### ISSUE TITLE
pyproject.toml in parent directory isn't examined

### ISSUE THREAD

**Issue by danra** [astral-sh/ruff-action#164]
I have my `pyproject.toml` in the parent directory of the one set to be checked by ruff. It's also the root directory of the repo.
With that setup, I expected setting `src` to the folder I'd like to check would still correctly find `pyproject.toml`, but apparently it doesn't:

```
Run astral-sh/ruff-action@v3
  with:
    src: ./python
    args: check
    github-token: ***
  env:
    PYTHONUNBUFFERED: true
    RUFF_OUTPUT_FORMAT: github
Could not find python/pyproject.toml. Using latest version.
Found ruffDir in tool-cache for 0.11.13
Added /opt/hostedtoolcache/ruff/0.11.13/x86_64 to the path
Set RUFF_OUTPUT_FORMAT to github
Successfully installed ruff version 0.11.13
/opt/hostedtoolcache/ruff/0.11.13/x86_64/ruff check ./python
```

I noticed this when I got unexpected ruff warnings in CI, which was due to the newest ruff version being used rather than the one specified in `pyproject.toml` (0.11.9).

Since `ruff` itself looks in parent directories for `pyproject.toml` when invoked, I think it would make sense for this to work for the action as well.

**Comment by eifinger** [comment]
For now you can work around that by using https://github.com/astral-sh/ruff-action?tab=readme-ov-file#install-a-version-from-a-specified-version-file

**Comment by lbrealdev** [comment]
I'm receiving a warning message similar when performing my CI workflow, but here the message is:

```shell
Warning: Could not parse version from pyproject.toml. Using latest version.
```

**Comment by lbrealdev** [comment]
> I'm receiving a warning message similar when performing my CI workflow, but here the message is:
> 
> Warning: Could not parse version from pyproject.toml. Using latest version.
> I don't know if my case fits in this issue, let me know if a new issue need to be created.
> 
> **NOTE:** In fact, my CI workflow works well, but I am seeing this warning in all executions.

Some updates .... I was able to make this warning disappear.

I was using `ruff-action` as follows:
```yaml
      - uses: astral-sh/ruff-action@v3
```

So, doing some tests, I added the input `version` and the warning message disappeared.
```yaml
      - uses: astral-sh/ruff-action@v3
        with:
          version: "latest"
```

Thanks!

**Comment by eifinger** [comment]
> > I'm receiving a warning message similar when performing my CI workflow, but here the message is:
> > Warning: Could not parse version from pyproject.toml. Using latest version.
> > I don't know if my case fits in this issue, let me know if a new issue need to be created.
> > **NOTE:** In fact, my CI workflow works well, but I am seeing this warning in all executions.
> 
> Some updates .... I was able to make this warning disappear.
> 
> I was using `ruff-action` as follows:
> 
>       - uses: astral-sh/ruff-action@v3
> 
> So, doing some tests, I added the input `version` and the warning message disappeared.
> 
>       - uses: astral-sh/ruff-action@v3
>         with:
>           version: "latest"
> 
> Thanks!

I fixed this in https://github.com/astral-sh/ruff-action/pull/209. Will be included in the next release

**Comment by DhavalGojiya** [comment]
#### Documentation Improvement Suggestion

Reference: [ruff-action › Install specific versions](https://github.com/astral-sh/ruff-action?tab=readme-ov-file#install-specific-versions)

> By default, this action looks for a pyproject.toml file in the root of the repository to determine the Ruff version to install.
If no pyproject.toml file is found, or no Ruff version is defined in project.dependencies, project.optional-dependencies, or dependency-groups, the latest version is installed.

---

#### Suggestion

The documentation could be improved.
It currently states that the action always checks the root of the repository for pyproject.toml.
However, if the src input is provided and does not point to the current directory ("."), this action will currently check pyproject.toml in that given src directory.

If parent of parents (pyproject.toml) checking logic implemented then may be need to update this Readme as well.
